src/modes/top10/Top10.svelte

```diff
@@ -174,15 +175,44 @@
     return data;
   });
 
+  // Compute the neighborhood around the current date.
```

Comment:
I'm not sure whether is as reactive as it should be. 

to my knowledge the `const date` won't be reactive, the `$: neighboorhood` would be but just uses non-reactive variables. 

Have you considered extracting this whole logic to a single function as in:

```js
$: neighborhood = computeNeighborhood($currentDateObject, $smallMultipleTimeSpan);
```

Reply:
regarding the neighboorhood computation:

https://github.com/cmu-delphi/www-covidcast/blob/c150bb3ca729ae618b927363d7187019499cde6f/src/stores/index.js#L124-L134

computes the small multiple time span which limits the view to four months around the current date. As far as I understand the neighborhood does a similar thing but for around 1 month. Can you extract the logic of both function into a utility function having a date and the offset as parameter and it will return the date range.

Reply:
> I'm not sure whether is as reactive as it should be.
> 
> to my knowledge the `const date` won't be reactive, the `$: neighboorhood` would be but just uses non-reactive variables.

That sounds right.  I'm not clear yet on what needs to be reactive.   But since the $currentDateObject is never changed in the top-10 page, the neighborhood would also not need to be reactive.  However, it seems like we _should_ allow the $currentDateObject to be changed.   

> 
> Have you considered extracting this whole logic to a single function as in:
> 
> ```js
> $: neighborhood = computeNeighborhood($currentDateObject, $smallMultipleTimeSpan);
> ```

That sounds like a good idea.  Although $smallMultipleTimeSpan turns out to not be precisely relevant on the top-10 page since it is a subset of the top-10 timespan.

Reply:
> regarding the neighboorhood computation:  [...]
> 
> computes the small multiple time span which limits the view to four months around the current date. As far as I understand the neighborhood does a similar thing but for around 1 month. 

Well, in prior discussions, we talked about a neighborhood of a couple weeks or months, but it seems more general and useful to use a neighborhood that is a fraction of the top-10 timespan, however long that is.  That way, if we allow zooming or panning, or other ways of selecting the total timespan, the neighborhood should always scale to the current timespan.

But I couldn't find a way to determine the top-10 timespan, other than by iterating through potentially all the data, given that some data could be missing.   Is there a way?  Alternatively, we could _always_ use the $smallMultipleTimeSpan, which has the advantage of always being consistent with what we show in small multiples in other places.

> Can you extract the logic of both function into a utility function having a date and the offset as parameter and it will return the date range.

This sounds like it would be a useful function in any event, though there is a slightly different purpose here.  The small multiple time span is always equally 1 month around the current date, or shifted back if needed.  But this neighborhood calculation was intended to be a subset of whatever the current data timespan is, so we may have to shift back (if overlapping the end of the timespan) or forward (if overlapping the start of the timespan).  I'll see if I can generalize this, however.